File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.py
class Solution:
    def rearrangeArray(self, nums: List[int]) -> List[int]:
        res=[0]*len(nums)
        pos=0
        neg=1
        for i in nums:
            if i<0:
                res[neg]=i
                neg+=2
            else:
                res[pos]=i
                pos+=2
        return res        
                

File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.cpp
class Solution {
public:
    vector<int> rearrangeArray(vector<int>& nums) {
        int s=nums.size();
        int pos=0;
        int neg=1;
        vector<int>res(s,0);
            for(int i=0;i<s;i++){
                if(nums[i]<0){
                    res[neg]=nums[i];
                    neg=neg+2;
                }
                else{
                    res[pos]=nums[i];
                        pos=pos+2;
                }
            }
        return res;
    }
};